chemprop/train/train.py

```diff
@@ -50,65 +52,75 @@ def train(model: MoleculeModel,
+        targets = torch.tensor([[0 if x is None else x for x in tb] for tb in target_batch])  # shape(batch, tasks)
 
         if args.target_weights is not None:
             target_weights = torch.tensor(args.target_weights).unsqueeze(0) # shape(1,tasks)
         else:
             target_weights = torch.ones(targets.shape[1]).unsqueeze(0)
-        data_weights = torch.tensor(data_weights_batch).unsqueeze(1) # shape(batch,1)
+        data_weights = torch.tensor(data_weights_batch).unsqueeze(1)  # shape(batch,1)
 
-        if args.loss_function == 'bounded_mse':
-            lt_target_batch = batch.lt_targets() # shape(batch, tasks)
-            gt_target_batch = batch.gt_targets() # shape(batch, tasks)
+        if args.loss_function == "bounded_mse":
+            lt_target_batch = batch.lt_targets()  # shape(batch, tasks)
+            gt_target_batch = batch.gt_targets()  # shape(batch, tasks)
             lt_target_batch = torch.tensor(lt_target_batch)
             gt_target_batch = torch.tensor(gt_target_batch)
 
         # Run model
         model.zero_grad()
-        preds = model(mol_batch, features_batch, atom_descriptors_batch, atom_features_batch, bond_features_batch)
+        preds = model(
+            mol_batch,
+            features_batch,
+            atom_descriptors_batch,
+            atom_features_batch,
+            bond_features_batch,
+        )
 
         # Move tensors to correct device
         torch_device = preds.device
         mask = mask.to(torch_device)
         targets = targets.to(torch_device)
         target_weights = target_weights.to(torch_device)
         data_weights = data_weights.to(torch_device)
-        if args.loss_function == 'bounded_mse':
+        if args.loss_function == "bounded_mse":
             lt_target_batch = lt_target_batch.to(torch_device)
             gt_target_batch = gt_target_batch.to(torch_device)
 
         # Calculate losses
-        if args.loss_function == 'mcc' and args.dataset_type == 'classification':
-            loss = loss_func(preds, targets, data_weights, mask) *target_weights.squeeze(0)
-        elif args.loss_function == 'mcc': # multiclass dataset type
+        if args.loss_function == "mcc" and args.dataset_type == "classification":
+            loss = loss_func(preds, targets, data_weights, mask) * target_weights.squeeze(0) * mask.sum()
+        elif args.loss_function == "mcc":  # multiclass dataset type
             targets = targets.long()
             target_losses = []
             for target_index in range(preds.size(1)):
                 target_loss = loss_func(preds[:, target_index, :], targets[:, target_index], data_weights, mask[:, target_index]).unsqueeze(0)
                 target_losses.append(target_loss)
-            loss = torch.cat(target_losses).to(torch_device) * target_weights.squeeze(0)
-        elif args.dataset_type == 'multiclass':
+            loss = torch.cat(target_losses).to(torch_device) * target_weights.squeeze(0) * mask.sum()
```

Comment:
```suggestion
            loss = torch.cat(target_losses) * target_weights.squeeze(0) * mask.sum()
```
same comment as above with the addition that the tensor should already be on the correct device